internal/configs/virtualserver_test.go

```diff
@@ -287,9 +287,14 @@ func TestGenerateVirtualServerConfig(t *testing.T) {
 	isPlus := false
 	isResolverConfigured := false
 	tlsPemFileName := ""
-	result := generateVirtualServerConfig(&virtualServerEx, tlsPemFileName, &baseCfgParams, isPlus, isResolverConfigured)
+	vsc := newVirtualServerConfigurator(&baseCfgParams, isPlus, isResolverConfigured)
+	result, warnings := vsc.GenerateVirtualServerConfig(&virtualServerEx, tlsPemFileName)
 	if !reflect.DeepEqual(result, expected) {
-		t.Errorf("generateVirtualServerConfig returned \n%v but expected \n%v", result, expected)
+		t.Errorf("GenerateVirtualServerConfig returned \n%v but expected \n%v", result, expected)
+	}
+
+	if len(warnings) != 0 {
+		t.Errorf("GenerateVirtualServerConfig returned warnings for %v", virtualServerEx)
```

Comment:
I suggest printing warnings instead of virtualServerEx, otherwise we don't know what warnings were generated. Also, `virtualServerEx` is a big structure and it is defined in the test. no need to print it. 

same suggestions for line 519, 781.